### ISSUE TITLE
Average value metric

### ISSUE THREAD

**Issue by vfdev-5** [pytorch/ignite#512]
Discussed on the slack:
> Joost van Amersfoort [7:14 PM]
I have a very simple question, how do I compute an average over a validation set?I have an evaluator engine and I am considering the EpochMetric, and pass `None` as the label and make the compute_fn only average over the first column, or the `RunningAverage` metric with `alpha=1`. However both feel a bit gross for this very simple problem.

> Interesting, looking at the `Loss` source then it indeed seems that it’s very close to what I want and more scalable than EpochMetric. But the problem is that it’s setup for supervised learning and forces len(output) >= 2

Idea is to provide some metric class similar to `Loss` but with no restrictions, just to compute average on provided iteration values.

**Comment by y0ast** [comment]
Thanks for making this issue @vfdev-5 

Would it be reasonable to just remove the restrictions from Loss? Perhaps make arguments for supervised learning the default arguments.

**Comment by vfdev-5** [comment]
> Perhaps make arguments for supervised learning the default arguments.

@y0ast should we still compute a `loss_fn` on a single entry ? 
https://github.com/pytorch/ignite/blob/fc85e25dc4f938d780b4c425acb2d40f6cac6f24/ignite/metrics/loss.py#L44

**Comment by y0ast** [comment]
In my case, `y_pred` is batch size by 1. So my loss function is just `torch.mean` to average over the batch.

It's a bit ugly to call that a loss function, so perhaps a different metric is a good idea after all.
